src/SonataPageAdminTrait.php

```diff
@@ -0,0 +1,276 @@
+    {
+        $tab = $this->getSession()->getPage()->find('css', sprintf('li.page-composer__container__child:contains(\'%s\') * a:contains(\'%s\')', $block, $tab));
+
+        if (null === $tab) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'tab', $tab);
+        }
+
+        $tab->click();
+    }
+
+    /**
+     * I should see n blocks.
+     *
+     * @When /^I should see (\d+) blocks$/
+     *
+     * @param int $num
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iShouldSeeBlocks(int $num): void
+    {
+        $selector = 'div.page-composer__container__view > ul.page-composer__container__children > li';
+
+        $element = $this->getSession()->getPage()->find('css', $selector);
+
+        if (null === $element) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'elements', $element);
+        }
+
+        $this->assertNumElements($num, $selector);
+    }
+
+    /**
+     * I open the block.
+     *
+     * @When /^I open the block "([^"]*)"$/
+     *
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iOpenTheBlock(string $name): void
+    {
+        $element = sprintf('li.page-composer__container__child:contains(\'%s\') > a.page-composer__container__child__edit', $name);
+
+        $this->iWaitForCssElementBeingVisible($element, 2);
+
+        $block = $this->getSession()->getPage()->find('css', $element);
+
+        if (null === $block) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $block);
+        }
+
+        $block->click();
+    }
+
+    /**
+     * I submit the block.
+     *
+     * @When /^I submit the block "([^"]*)"$/
+     *
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iSubmitTheBlock(string $name): void
+    {
+        $button = $this->getSession()->getPage()->find('css', sprintf('li.page-composer__container__child:contains(\'%s\') * button[type=submit]', $name));
+
+        if (null === $button) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $button);
+        }
+
+        $button->press();
+    }
+
+    /**
+     * I delete the block.
+     *
+     * @When /^I delete the block "([^"]*)"$/
+     *
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iDeleteTheBlock(string $name): void
+    {
+        $block = $this->getSession()->getPage()->find('css', sprintf('li:contains(\'%s\')', $name));
+
+        if (null === $block) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $block);
+        }
+
+        $id = $block->getAttribute('data-block-id');
+
+        $this->visit(sprintf('admin_app_sonata_block_delete;id=%s', $id));
```

Comment:
maybe it would be better to open the block and click on delete instead of accessing to the route directly no? it's not especially better but it's what a user really does on front so i think it would be better to be as close as possible from front interaction